src/Context/ApiContext.php

```diff
@@ -273,15 +273,19 @@ private function dropSessionContext()
      */
     public function ensureSessionActive()
     {
-        if (is_null($this->sessionContext)) {
-            return;
+        if (!is_null($this->sessionContext) && $this->isSessionExpired()) {
+            $this->resetSession();
         }
+    }
 
+    /**
+     * @return bool
+     */
+    public function isSessionExpired(): bool
+    {
         $timeExpiry = $this->sessionContext->getExpiryTime()->getTimestamp();
 
-        if ($timeExpiry - time() < self::TIME_TO_SESSION_EXPIRY_MINIMUM_SECONDS) {
-            $this->resetSession();
-        }
+        return is_null($this->sessionContext) || $timeExpiry - time() < self::TIME_TO_SESSION_EXPIRY_MINIMUM_SECONDS;
```

Comment:
Since you do `is_null($this->sessionContext)` check here, the method can be be opposite to what we've had and be called `isSessionActive` or `hasActiveSession`. This also fits well into our `ensureSessionActive` :)

Also, I'd go for an explanatory variable here. I missed it when initially built `ensureSessionActive` %).

So something like this:

```
if (is_null($this->sessionContext)) {
    return false;
} else {
    $timeExpiry = $this->sessionContext->getExpiryTime()->getTimestamp();
    $timeToExpiry = $timeExpiry - time();

    return self::TIME_TO_SESSION_EXPIRY_MINIMUM_SECONDS < $timeToExpiry;
}
```

Reply:
Bueno